src/pretix/presale/views/event.py

```diff
@@ -104,6 +108,29 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class EventIcalDownload(EventViewMixin, View):
+    def get(self, request, *args, **kwargs):
+        if not self.request.event:
+            raise Http404(_('Unknown event code or not authorized to access this event.'))
+
+        cal = Calendar()
+        cal.add('version', '2.0')
+        cal.add('prodid', '-//{}//{}//'.format(self.request.event.organizer.name, self.request.event.name))
+
+        event = Event()
+        event.add('summary', self.request.event.name)
+        event.add('dtstart', self.request.event.date_from)
```

Comment:
> However, you also mentioned timezone. My bad :( The original version is wrong!
> Now I fixed it and it will output

You did not upload that fix yet, did you? Timezones are always nasty :(

> I would like to commit my changes after confirming the behaviours of the event settings.
> For `self.request.event.settings.show_times`, it's easy since `icalendar` supports date object too!

Nice! Can you confirm that it then correctly outputs a full-day (or multiple-day) event?

> So if `self.request.event.settings.show_times is True and self.request.event.settings.show_date_to is False`,
I guess we can just provide DTSTART so that the time can be kept.

Yes, this seems like the correct thing, at least when looking at the spec. I have my doubts on how calendar applications out there will handle this, but let's go for this version for now.

> If `self.request.event.settings.show_times is False and self.request.event.settings.show_date_to is False`,
we will set `DTSTART` date and the event lasts for one day.

Seems about right.

> If `self.request.event.settings.show_times is False and self.request.event.settings.show_date_to is True`, we will ignore timezone and set `date` in both `DTSTART` and `DTEND`.

Yep.

> Does it sound good?

Yes it does! :) Just go ahead and tell me if you need any more help.